### DISCUSSION TITLE
[Question] Overwrite CartStockAvailabilityValidator

### DISCUSSION THREAD

**Discussion by lukadschaak** [coreshop/CoreShop#2420] [Q&A]
| Q                | A
| ---------------- | -----
| Bug report?      | no
| Feature request? | no
| BC Break report? | no
| RFC?             | no
| Question | yes

Moin!
We are currently working on overriding some validators for the checkout. For various reasons. In one case we have a sort of sub items whose stock we also want to validate. So we would like to override the `CartStockAvailabilityValidator`.

What is the best way to do this? In the checkout steps, the order is validated in its entirety, and the contraints for this are configured here.
```yaml
# src/CoreShop/Bundle/CoreBundle/Resources/config/validation/Order.yml
CoreShop\Component\Core\Model\Order:
    constraints:
        - \CoreShop\Bundle\CoreBundle\Validator\Constraints\CartStockAvailability: { groups: ['coreshop'] }
        - \CoreShop\Bundle\CoreBundle\Validator\Constraints\CartMinimumQuantity: { messageBelowMinimum: 'coreshop.cart_item.quantity_below_min_limit', groups: ['coreshop'] }
        - \CoreShop\Bundle\CoreBundle\Validator\Constraints\CartMaximumQuantity: { messageAboveMaximum: 'coreshop.cart_item.quantity_above_max_limit', groups: ['coreshop'] }
```

Unfortunately, we cannot replace validators. The configuration can only be extended, but not overwritten. Do we have to decorate the class now or what would be the best practice?

**Comment by dpfaffenbauer** [comment]
@lukadschaak 
1. Create your own Validator and validate how you need it.
2. Create your own Validation Group and change the CoreShop Validation Group to your new one.
3. Copy the CoreShop Validation Files and replace the coreshop validation group with yours

**Comment by lukadschaak** [comment]
@dpfaffenbauer Thank for your fast answer. Some thoughts for future reader:

> 1. Create your own Validator and validate how you need it.

Good suolution over all, but a bit unintuitive because aliases must be used.

> 2. Create your own Validation Group and change the CoreShop Validation Group to your new one.

Code must be overwritten for this approach. Not our favorite solution.

> 3. Copy the CoreShop Validation Files and replace the coreshop validation group with yours

If your mean the config files (.yml), then as far as i know this is not possible for validation: https://symfony.com/doc/current/bundles/override.html#validation-metadata

**Reply by dpfaffenbauer** [reply]
right, so then only 1 and 2 is the way to go.
